# NBA Hackathon New York City Appication
## By  Zhongheng Li - heng2j@nyu.edu 





## 2) BONUS: For an upcoming presentation, you are asked to predict total gate revenue for the entire NBA Playoffs. You should use the win probabilities data provided in this file and the Hypothetical Playoff Gate Data tab in this file to answer part b of the question.

### a) As a primer, let’s think about an individual series. Team East1 plays Team East8 in the First Round. Suppose the probability that East1 defeats East8 at home (8 @ 1) is p_H, and the probability that East1 defeats East8 on the road (1 @ 8) is p_A. In terms of p_H and p_A, what is the probability that the series goes exactly N games for N in {4,5,6,7}, regardless of winner? Please give closed-form analytical solutions.

### Answer:

In [ ]:
Listing all possible combinations in closed-form:

N = 4 Win( 4:0 ) or Lost( 0:4 )

    Play 4 games: p_H * p_H * p_A * p_A 

   Answer:
    (p_H** ) * (p_A** ) +  (1 - p_H)**  * (1 - p_A) **
    



N = 5  Win( 4:1 ) or  Lost(1:4)

    Play 5 games: p_H * p_H * p_A * p_A * p_H
        
    Total combinatins:
    4C3 = 4
    
    Sub combinations:
    3C2 = 3
    
    
   Answer:
   ((1 - p_H) * p_H * p_A * p_A * p_H + 
    p_H * (1 - p_H) * p_A * p_A * p_H + 
    p_H * p_H * (1 - p_A) * p_A * p_H + 
    p_H * p_H * p_A * (1 - p_A) * p_H )
    
    + 
    
   (p_H * (1 - p_H) * (1 - p_A) * (1 - p_A) * (1 - p_H) + 
    (1 - p_H) * p_H * (1 - p_A) * (1 - p_A) * (1 - p_H) + 
    (1 - p_H) * (1 - p_H) * p_A * (1 - p_A) * (1 - p_H) + 
    (1 - p_H) * (1 - p_H) * (1 - p_A) * p_A * (1 - p_H)) 
    

    
N = 6 ( 4:2 ) or (2:4)
   
    Play 6 games: p_H * p_H * p_A * p_A * p_H * p_A
        
    Total combinatins:  
    5C3  
    5! / (3! * 2!) = 10 
    
    Sub combinations:
    3C2 = 3
        
   Answer:    
   ((1 - p_H) * (1 - p_H) * p_A * p_A * p_H * p_A + 
   
    p_H * (1 - p_H) * (1 - p_A) * p_A * p_H * p_A + 
    p_H * (1 - p_H) * p_A * (1 - p_A) * p_H * p_A + 
    p_H * (1 - p_H) * p_A * p_A * (1 - p_H) * p_A + 
    
    p_H * p_H * (1 - p_A) * (1 - p_A) * p_H * p_A + 
    p_H * p_H * p_A * (1 - p_A) * (1 - p_H) * p_A + 
    p_H * p_H * (1 - p_A) * p_A * (1 - p_H) * p_A + 
   
    (1 - p_H) * p_H * (1 - p_A) * p_A * p_H * p_A + 
    (1 - p_H) * p_H * p_A * (1 - p_A) * p_H * p_A + 
    (1 - p_H) * p_H * p_A * p_A  * (1 - p_H) * p_A ) 

    
    +  
    
   ( p_H * p_H * (1 - p_A) * (1 - p_A) * (1 - p_H) * (1 - p_A) + 
   
     p_H * (1 - p_H) * p_A * (1 - p_A) * (1 - p_H) * (1 - p_A) + 
     p_H * (1 - p_H) * (1 - p_A) * p_A * (1 - p_H) * (1 - p_A) + 
     p_H * (1 - p_H) * (1 - p_A) * (1 - p_A) * p_H * (1 - p_A) + 
   
     (1 - p_H) * (1 - p_H) * p_A * p_A * (1 - p_H) * (1 - p_A) + 
     (1 - p_H) * (1 - p_H) * (1 - p_A) * p_A * p_H * (1 - p_A) + 
     (1 - p_H) * (1 - p_H) * p_A * (1 - p_A) * p_H * (1 - p_A) + 
     
     (1 - p_H) * p_H * (1 - p_A) * p_A * (1 - p_H) * (1 - p_A) + 
     (1 - p_H) * p_H * p_A * (1 - p_A) * (1 - p_H) * (1 - p_A) + 
     (1 - p_H) * p_H * (1 - p_A) * (1 - p_A) * p_H * (1 - p_A) ) 
    
    
        
N = 7 ( 4:3 ) or (3:4)
    p_H * p_H * p_A * p_A * p_H * p_A * p_H
    
    Total combinatins:
    6C3
    6! / (3!* 3!) = 20
    
    Sub combinations:
    3C2 = 3
    
   Answer:
   (p_H * p_H * p_A * (1 - p_A) * (1 - p_H) * (1 - p_A) * p_H +
    (1 - p_H) * (1 - p_H) * (1 - p_A) * p_A * p_H * p_A * p_H +
   
    p_H * (1 - p_H) * (1 - p_A) * (1 - p_A) * p_H * p_A * p_H + 
    p_H * (1 - p_H) * (1 - p_A) * p_A * (1 - p_H) * p_A * p_H +
    p_H * (1 - p_H) * (1 - p_A) * p_A * p_H * (1 - p_A) * p_H +
   
    p_H * (1 - p_H) * p_A * (1 - p_A) * p_H * (1 - p_A) * p_H +
    p_H * (1 - p_H) * p_A * p_A * (1 - p_H) * (1 - p_A) * p_H +
    p_H * (1 - p_H) * p_A * (1 - p_A) * (1 - p_H) * p_A * p_H +
    
    p_H * p_H * (1 - p_A) * (1 - p_A) * (1 - p_H) * p_A * p_H +
    p_H * p_H * (1 - p_A) * (1 - p_A) * p_H * (1 - p_A) * p_H +
    p_H * p_H * (1 - p_A) * p_A * (1 - p_H) * (1 - p_A) * p_H +
    
    (1 - p_H) * (1 - p_H) * p_A * p_A * (1 - p_H) * p_A * p_H +
    (1 - p_H) * (1 - p_H) * p_A * p_A * p_H * (1 - p_A) * p_H +
    (1 - p_H) * (1 - p_H) * p_A * (1 - p_A) * p_H * p_A * p_H +

    (1 - p_H) * p_H * (1 - p_A) * p_A * p_H * (1 - p_A) * p_H +
    (1 - p_H) * p_H * (1 - p_A) * (1 - p_A) * p_H * p_A * p_H +
    (1 - p_H) * p_H * (1 - p_A) * p_A * (1 - p_H) * p_A * p_H +
    
    (1 - p_H) * p_H * p_A * p_A * (1 - p_H) * (1 - p_A) * p_H +
    (1 - p_H) * p_H * p_A * (1 - p_A) * (1 - p_H) * p_A * p_H +
    (1 - p_H) * p_H * p_A * (1 - p_A) * p_H * (1 - p_A) * p_H )
    
    +
    
   (p_H * p_H * p_A * (1 - p_A) * (1 - p_H) * (1 - p_A) * (1 - p_H) +
    (1 - p_H) * (1 - p_H) * (1 - p_A) * p_A * p_H * p_A * (1 - p_H) +
  
    p_H * (1 - p_H) * (1 - p_A) * (1 - p_A) * p_H * p_A * (1 - p_H) + 
    p_H * (1 - p_H) * (1 - p_A) * p_A * (1 - p_H) * p_A * (1 - p_H) +
    p_H * (1 - p_H) * (1 - p_A) * p_A * p_H * (1 - p_A) * (1 - p_H) +
   
    p_H * (1 - p_H) * p_A * (1 - p_A) * p_H * (1 - p_A) * (1 - p_H) +
    p_H * (1 - p_H) * p_A * p_A * (1 - p_H) * (1 - p_A) * (1 - p_H) +
    p_H * (1 - p_H) * p_A * (1 - p_A) * (1 - p_H) * p_A * (1 - p_H) +
    
    p_H * p_H * (1 - p_A) * (1 - p_A) * (1 - p_H) * p_A * (1 - p_H) +
    p_H * p_H * (1 - p_A) * (1 - p_A) * p_H * (1 - p_A) * (1 - p_H) +
    p_H * p_H * (1 - p_A) * p_A * (1 - p_H) * (1 - p_A) * (1 - p_H) +
    
    (1 - p_H) * (1 - p_H) * p_A * p_A * (1 - p_H) * p_A * (1 - p_H) +
    (1 - p_H) * (1 - p_H) * p_A * p_A * p_H * (1 - p_A) * (1 - p_H) +
    (1 - p_H) * (1 - p_H) * p_A * (1 - p_A) * p_H * p_A * (1 - p_H) +

    (1 - p_H) * p_H * (1 - p_A) * p_A * p_H * (1 - p_A) * (1 - p_H) +
    (1 - p_H) * p_H * (1 - p_A) * (1 - p_A) * p_H * p_A * (1 - p_H) +
    (1 - p_H) * p_H * (1 - p_A) * p_A * (1 - p_H) * p_A * (1 - p_H) +
    
    (1 - p_H) * p_H * p_A * p_A * (1 - p_H) * (1 - p_A) * (1 - p_H) +
    (1 - p_H) * p_H * p_A * (1 - p_A) * (1 - p_H) * p_A * (1 - p_H) +
    (1 - p_H) * p_H * p_A * (1 - p_A) * p_H * (1 - p_A) * (1 - p_H) 
     

### Convert this answer into python functions for future questions:

In [1]:
def getProb_In4( input_p_H, input_p_A , winOrLose_str ):
     
    if(winOrLose_str == 'lose'):
        p_H = (1 - input_p_H)
        p_A = (1 - input_p_A)
    else: 
        p_H = input_p_H
        p_A = input_p_A
        
    return (p_H**2 ) * (p_A**2 )


In [2]:
## Testing with East1 vs East8
getProb_In4(0.978,0.942,'win') + getProb_In4(0.978,0.942,'lose') 

0.848751096352

In [3]:
def getProb_In5( p_H, p_A , winOrLose_str ):
     
    result = 0
    
    if(winOrLose_str == 'lose'):
    
        result = 2 * (p_H * (1 - p_A)**2  *  (1 - p_H)**2) + 2 * ((1 - p_H)**3 * p_A  *  (1 - p_A)) 
        

    else: 
    
       result = 2 * ((1 - p_H) * p_H**2 * p_A**2) +  2 * (p_H**3 * (1 - p_A) * p_A)  
    
    return result



In [4]:
## Testing with East1 vs East8
getProb_In5(0.978,0.942,'win') + getProb_In5(0.978,0.942,'lose') 

0.1395668722560001

In [5]:
def getProb_In6( p_H, p_A , winOrLose_str ):
     
    result = 0
    
    if(winOrLose_str == 'lose'):
    
        result =  ( 2 * (p_H**2 * (1 - p_A)**3 * (1 - p_H)) + 2 * (p_H * (1 - p_H)**2 * p_A * (1 - p_A)**2 ) + (1 - p_H)**3 * p_A**2  * (1 - p_A) + (1 - p_H) * p_H**2 * (1 - p_A)**2 + 2 * ((1 - p_H)**2 * (1 - p_A)**2 * p_A * p_H)  + 2 * ((1 - p_H)**2 * p_H * (1 - p_A)**2 * p_A) ) 
 
    else: 
    
       result =  (2* ((1 - p_H)**2 * p_A**3 * p_H) + 2* (p_H**2 * (1 - p_H) * p_A**2 * (1 - p_A)) + p_A**3 * (1 - p_H)**2 * p_H + p_H**3 * (1 - p_A)**2 * p_A + 2* (p_H**2 * p_A**2 * (1 - p_A) * (1 - p_H)) + 2*((1 - p_H) * p_H**2 * p_A**2 * (1 - p_A))) 

    
    return result



In [6]:
## Testing with East1 vs East8
getProb_In6(0.978,0.942,'win') + getProb_In6(0.978,0.942,'lose') 

0.010737902027386963

In [7]:
def getProb_In7( p_H, p_A , winOrLose_str ):
     
    result = 0
    
    if(winOrLose_str == 'lose'):
    
        result =  (p_H**2 * p_A * (1 - p_A)**2 * (1 - p_H)**2  + (1 - p_H)**3 * (1 - p_A) * p_A**2 * p_H  + 2 * (p_H**2 * (1 - p_H)**2 * (1 - p_A)**2 * p_A) +  2 * (p_H * (1 - p_H)**3 * p_A**2 * (1 - p_A))  + 2* (p_H**2 * (1 - p_A)**2 * (1 - p_H)**2 * p_A)  + p_H**3 * (1 - p_A)**3 * (1 - p_H) + (1 - p_H)**4 * p_A**3  + 2 * ((1 - p_H)**3 * p_A**2 * p_H * (1 - p_A))+ 2 * ((1 - p_H)**2 * p_H**2 * (1 - p_A)**2 * p_A)   + 2 * (p_H * (1 - p_H)**3 * (1 - p_A) * p_A**2) +  2 * (p_H**2 * (1 - p_H)**2 * p_A * (1 - p_A)**2)  + 2 * ((1 - p_H)**3 * p_H * p_A**2 * (1 - p_A)))    
    
    else: 
    
       result = (p_H**3 * p_A * (1 - p_A)**2 * (1 - p_H) + (1 - p_H)**3  * p_A**3 * p_H + p_H**4 * (1 - p_A)**3 +(1 - p_H)**2 * (1 - p_A) * p_A**2 * p_H**2 + 2* (p_H**3 * (1 - p_H) * (1 - p_A)**2 * p_A) + 2*(p_H**2 * (1 - p_H)**2 * (1 - p_A) * p_A**2) + 2* (p_H**3 * (1 - p_H) * p_A * (1 - p_A)**2) + 2* (p_H**2 * (1 - p_H)**2 * p_A**2 * (1 - p_A))  + 2 * (p_H**3  * (1 - p_A)**2 * p_A * (1 - p_H))  + 2 * ((1 - p_H)**2 * p_A**2 * (1 - p_A) * p_H**2) + 2 * ((1 - p_H) * (1 - p_A)**2 * p_A * p_H**3)   + 2 * ((1 - p_H)**2 * p_H**2 * p_A**2 * (1 - p_A)))


    
    return result



In [8]:
## Testing
getProb_In7(0.978,0.942,'win') + getProb_In7(0.978,0.942,'lose') 

0.0010108111593484827

In [9]:
def getTotalProb_WinSeries ( p_H, p_A ):
    
    result = getProb_In4(p_H, p_A , 'win') + getProb_In5(p_H, p_A , 'win') + getProb_In6(p_H, p_A , 'win') + getProb_In7(p_H, p_A , 'win')
    
    return result


In [10]:
## Testing with East1 vs East8
getTotalProb_WinSeries ( .978,0.942 )

0.9999499206999286

## b) Now suppose that the regular season has just concluded and the NBA playoff picture is set. Team Finance has forecasted gate revenue per game for every team in each possible round – these per-game forecasts for each (team, round) combination are given in the Hypothetical Playoff Gate Data tab in the attached Excel file. Further suppose you are given win_probabilities.csv, which describes the probability that a given team defeats any other team, either at home or on the road. (The format of the probabilities in each row may not be standard.) Note that there is no re-seeding in the NBA Playoffs and that each series follows a 2-2-1-1-1 format. If equal seeds make the Finals, suppose the West team has home court advantage; otherwise, suppose the better seed has home court. Build a playoff simulator that tracks gate revenue using this data. (Assume each game is independent.) 

### Goal: Build a playoff simulator that tracks gate revenue 

### Design & Planing:

In [ ]:
Thoughts on Requirements:
    1. Show probability of winning the current series / facing each seed / going to the final 
    2. Show money to earn for each series if win at Game N, for N in {4,5,6,7}
    
    
Steps:
    
    1. Winning percentage for each team on each series 
        Win r1, r2, r3, r4 
    
    2. Caculate the 1st round opponent based on the entered seed
        Default round 1 matches
        
      
        
    3. Wining Percentage for each series 
        Win = Avg(P(Win4) + P(Win5) +P(Win6)+P(Win7))
        
        

       
Case 1: When user enter East 1   
      
        1. Caculate the 1st round opponent based on the entered seed
             If opponent seed number is greater than East 1
                East 1 has Home Advantage then use East 1 p_H & p_A
        2. 

Output:
    Team  Round POfEncounter          POfWin       POfLose        POfPlaying_4G         ROf_4G                              POfPlaying_5G              ROf_5G                       POfPlaying_6G   ROf64G              POfPlaying_7G       ROf_7G                          
     E8      1     1                Avg(P(W))    1 - Avg(P(W))   p_H^2* p_A^2        r_H^2 *  r_A^2                                                 r_H^3 *  r_A^2                                  r_H^3 *  r_A^3                        r_H^4 *  r_A^3
     E5      2    E5 Win r1
     E4      2    E4 Win r1
     E7      3    E5 Win r1&r2
     E6      3    E6 Win r1&r2
     E3      3    E3 Win r1&r2
     E2      3    E3 Win r1&r2
        
     W1      4    W1 Win W(r1&r2&r3)
     W2      4    W2 Win W(r1&r2&r3)
     W3      4    W3 Win W(r1&r2&r3)
     W4      4    W4 Win W(r1&r2&r3)
     W5      4    W5 Win W(r1&r2&r3)
     W6      4    W6 Win W(r1&r2&r3)
     W7      4    W7 Win W(r1&r2&r3)
     W8      4    W8 Win W(r1&r2&r3)

        
 
Case 2: When user enter East 7

     If opponent seed number is less than East 7
        opponent has Home Advantage then use opponent (1 - p_H) & (1 - p_A)
     
     If opponent seed number is greater than East 7
        East 7 has Home Advantage then use East 7 p_H & p_A
     



Output:
    Team  Round POfEncounter          POfWin       POfLose        POfPlaying_4G                      RevenueOf_4G                              POfPlaying_5G              ROf_5G                       POfPlaying_6G   ROf64G              POfPlaying_7G       ROf_7G                          
     E2      1     1                Avg(P(W))    1 - Avg(P(W))   (1 - p_H)^2* (1 - p_A)^2         r_H^2 *  r_A^2                                                r_H^3 *  r_A^2                                  r_H^3 *  r_A^3                        r_H^4 *  r_A^3
     E8      3    E8 Win r1&r2
     E5      3    E5 Win r1&r2
     E6      2    E6 Win r1
     E3      2    E3 Win r1
     E4      3    E4 Win r1&r2
     E1      3    E1 Win r1&r2
        
     W1      4    f
     W2      4    W2 Win W(r1&r2&r3)
     W3      4    W3 Win W(r1&r2&r3)
     W4      4    W4 Win W(r1&r2&r3)
     W5      4    W5 Win W(r1&r2&r3)
     W6      4    W6 Win W(r1&r2&r3)
     W7      4    W7 Win W(r1&r2&r3)
     W8      4    W8 Win W(r1&r2&r3)
        
        
Case 1:When user enter East 1   
    
    Round 1: Opponent seed number
        East1 v East8
            East1 wins at Game N, for N in {4,5,6,7}
                N = 4 
                    p_H * p_H * p_A * p_A
                    2756446 * 2 + 1437278 * 2 = 8387448
               
                N = 5 
                    p_H * p_H * p_A * p_A * p_H
                    2756446 * 3 + 1437278 * 2 = 11143894
                
                N = 6   
                    p_H * p_H * p_A * p_A * p_H * p_A
                    
                    2756446 * 3 + 1437278 * 3 = 12581172
                    
                N = 7   
                    p_H * p_H * p_A * p_A * p_H * p_A * p_H
                    
                    2756446 * 4 + 1437278 * 3 = 15337618
                    
                
              

In [18]:
## Hardcoded ideal Matches to visualize the roundMatchesDict that is going to build
matches = {1: {'Round1':[8], 'Round2': [4,5] , 'Round3':[2,3,6,7]}, 
           2: {'Round1':[7], 'Round2': [3,6] , 'Round3':[1,4,5,8]},
           3: {'Round1':[6], 'Round2': [2,7] , 'Round3':[1,4,5,8]},
           4: {'Round1':[5], 'Round2': [1,8] , 'Round3':[2,3,6,7]},
           5: {'Round1':[4], 'Round2': [1,8] , 'Round3':[2,3,6,7]},
           6: {'Round1':[3], 'Round2': [2,7] , 'Round3':[1,4,5,8]},
           7: {'Round1':[2], 'Round2': [3,6] , 'Round3':[1,4,5,8]},
           8: {'Round1':[1], 'Round2': [4,5] , 'Round3':[2,3,6,7]}
          }

In [ ]:
## Envision the datastructure for Team Winning Probability Per Round
Winning Percentages for each series:
    
OutPut:
    
    Team    WinR1  WinR1*R2  WinR1*R2*R3
     E1     
     E2      
     E3
     E4
     E5
     E6
     E7
     E8

### Import modules

In [12]:
import numpy as np
import pandas as pd

from scipy.stats import entropy
import os
import json

from sklearn.model_selection import train_test_split
from sklearn.tree import DecisionTreeClassifier
from sklearn import tree
from sklearn.cross_validation import cross_val_score
from sklearn import metrics
from sklearn.metrics import roc_curve, auc
from sklearn.metrics import accuracy_score
from sklearn.svm import SVC
from sklearn import linear_model
from sklearn.metrics import accuracy_score
from decimal import *

import statsmodels.api as sm
from patsy import dmatrices
from sklearn.linear_model import LogisticRegression

from sklearn import preprocessing

import matplotlib.pyplot as plt
%matplotlib inline



from sklearn.ensemble import RandomForestClassifier
from sklearn import metrics

/Users/zhonghengli/anaconda/lib/python3.6/site-packages/sklearn/cross_validation.py:44: DeprecationWarning: This module was deprecated in version 0.18 in favor of the model_selection module into which all the refactored classes and functions are moved. Also note that the interface of the new CV iterators are different from that of this module. This module will be removed in 0.20.
  "This module will be removed in 0.20.", DeprecationWarning)
/Users/zhonghengli/anaconda/lib/python3.6/site-packages/statsmodels/compat/pandas.py:56: FutureWarning: The pandas.core.datetools module is deprecated and will be removed in a future version. Please use the pandas.tseries module instead.
  from pandas.core import datetools


### Load & Understand Data

In [13]:
win_probabilities_xlsx_file = pd.ExcelFile('data/win_probabilities.xlsx')
win_probabilities_xlsx_file.sheet_names

Business_Track_Application_Datasets_xlsx_file = pd.ExcelFile('data/Business-Track-Application-Datasets.xlsx')
Business_Track_Application_Datasets_xlsx_file

In [14]:
Hypothetical_Playoff_Gate_Data_df = Business_Track_Application_Datasets_xlsx_file.parse('Hypothetical Playoff Gate Data')

In [15]:
Hypothetical_Playoff_Gate_Data_df

Unnamed: 0 HomeTeam  Round1_Gate  Round2_Gate   Round3_Gate   Round4_Gate
0          NaN    East1      2756446   3528250.88  5.574636e+06  8.417701e+06
1          NaN    East2       722126    888214.98  1.163562e+06  1.838427e+06
2          NaN    East3       642409    732346.26  1.230342e+06  2.079278e+06
3          NaN    East4      3232910   4558403.10  7.384613e+06  1.270153e+07
4          NaN    East5      1780942   2546747.06  3.845588e+06  6.152941e+06
5          NaN    East6      1208557   1281070.42  2.216252e+06  3.435190e+06
6          NaN    East7      1026998   1201587.66  1.838429e+06  3.015024e+06
7          NaN    East8      1437278   2069680.32  2.814765e+06  4.728806e+06
8          NaN    West1      3624169   4928869.84  8.576234e+06  1.132231e+07
9          NaN    West2      1660090   1992108.00  2.848714e+06  4.814327e+06
10         NaN    West3      2856064   3313034.24  5.632158e+06  8.448237e+06
11         NaN    West4      2034910   2340146.50  2.948585e+06  5.071565e+06
12         NaN    West5      2522413   3279136.90  4.918705e+06  8.558547e+06
13         NaN    West6      2108399   2635498.75  4.216798e+06  6.325197e+06
14         NaN    West7      3100262   4433374.66  5.630386e+06  9.853175e+06
15         NaN    West8       974779   1276960.49  1.992058e+06  3.107611e+06

In [16]:
win_probabilities_df = win_probabilities_xlsx_file.parse('win_probabilities.csv')

In [389]:
win_probabilities_df.head(30)

Team1  Team2  Prob1WinsHome  Prob1WinsAway
0   East1  East2          0.580          0.517
1   East1  East3          0.672          0.630
2   East1  East4          0.718          0.668
3   East1  East5          0.785          0.710
4   East1  East6          0.819          0.788
5   East1  East7          0.936          0.893
6   East1  East8          0.978          0.942
7   East1  West1          0.482          0.467
8   East1  West2          0.630          0.563
9   East1  West3          0.660          0.619
10  East1  West4          0.740          0.674
11  East1  West5          0.790          0.709
12  East1  West6          0.840          0.795
13  East1  West7          0.910          0.839
14  East1  West8          0.960          0.916
15  East2  East3          0.616          0.567
16  East2  East4          0.675          0.580
17  East2  East5          0.760          0.686
18  East2  East6          0.790          0.758
19  East2  East7          0.866          0.823
20  East2  East8          0.896          0.858
21  East2  West1          0.483          0.459
22  East2  West2          0.553          0.481
23  East2  West3          0.610          0.568
24  East2  West4          0.649          0.631
25  East2  West5          0.684          0.659
26  East2  West6          0.811          0.748
27  East2  West7          0.874          0.840
28  East2  West8          0.929          0.879
29  East3  East4          0.600          0.554

In [29]:
## Set up Global Variables 
Seeds = [1,2,3,4,5,6,7,8] 

Rounds = ['R1', 'R2', 'R3']

In [20]:
## Set First Round matches Dict
roundMatchesDict = {}

len(Seeds) /2
index_h = 0
index_a = len(Seeds) - 1
while index_h < index_a:

    roundMatchesDict[Seeds[index_h]] = {}
    roundMatchesDict[Seeds[index_a]] = {}
    roundMatchesDict[Seeds[index_h]]['R1'] = []
    roundMatchesDict[Seeds[index_h]]['R1'].append(Seeds[index_a])
    
    roundMatchesDict[Seeds[index_a]]['R1'] = []
    roundMatchesDict[Seeds[index_a]]['R1'].append(Seeds[index_h])

    
    index_h = index_h + 1
    index_a = index_a - 1

In [22]:
roundMatchesDict

{1: {'R1': [8]},
 2: {'R1': [7]},
 3: {'R1': [6]},
 4: {'R1': [5]},
 5: {'R1': [4]},
 6: {'R1': [3]},
 7: {'R1': [2]},
 8: {'R1': [1]}}

In [23]:
## Set up Round 2 & 3 
for seed in Seeds:
    if(seed == 1) or (seed ==  8):
        roundMatchesDict[seed]['R2'] = [4,5]
        roundMatchesDict[seed]['R3'] = [2,3,6,7]
    elif(seed == 4) or (seed ==  5):
        roundMatchesDict[seed]['R2'] = [1,8]
        roundMatchesDict[seed]['R3'] = [2,3,6,7]
    elif(seed == 3) or (seed ==  6):
        roundMatchesDict[seed]['R2'] = [2,7]
        roundMatchesDict[seed]['R3'] = [1,4,5,8]
    elif(seed == 2) or (seed ==  7):
        roundMatchesDict[seed]['R2'] = [3,6]
        roundMatchesDict[seed]['R3'] = [1,4,5,8]
        


In [24]:
roundMatchesDict

{1: {'R1': [8], 'R2': [4, 5], 'R3': [2, 3, 6, 7]},
 2: {'R1': [7], 'R2': [3, 6], 'R3': [1, 4, 5, 8]},
 3: {'R1': [6], 'R2': [2, 7], 'R3': [1, 4, 5, 8]},
 4: {'R1': [5], 'R2': [1, 8], 'R3': [2, 3, 6, 7]},
 5: {'R1': [4], 'R2': [1, 8], 'R3': [2, 3, 6, 7]},
 6: {'R1': [3], 'R2': [2, 7], 'R3': [1, 4, 5, 8]},
 7: {'R1': [2], 'R2': [3, 6], 'R3': [1, 4, 5, 8]},
 8: {'R1': [1], 'R2': [4, 5], 'R3': [2, 3, 6, 7]}}

In [159]:
## Function to caculate the probability for each team from given conference to get to R4, the NBA Winning 
def getProbsToFinals(conference):
    
    
    for seed in Seeds:    

        thisSeedStr = conference + str(seed)

        TeamWinningPercentagePerRound[thisSeedStr] = {}

        totalWinProb = 0

        for r in Rounds:

            ## Calculate Winning Percentage for this round
            OpponentSeeds= roundMatchesDict[seed][r]
            OpponentSeedsLen = len(OpponentSeeds)

            round_winningProb = 0;

            for OpponentSeed in  OpponentSeeds:

                OpponentSeedStr = conference + str(OpponentSeed)

                if(seed > OpponentSeed):

                    probRow = win_probabilities_df.loc[(win_probabilities_df['Team1'] == OpponentSeedStr) & (win_probabilities_df['Team2'] == thisSeedStr)]

                    p_H = 1 - probRow.Prob1WinsHome
                    p_A = 1 - probRow.Prob1WinsAway


                else:

                    probRow = win_probabilities_df.loc[(win_probabilities_df['Team1'] == thisSeedStr) & (win_probabilities_df['Team2'] == OpponentSeedStr)]

                    p_H = probRow.Prob1WinsHome
                    p_A = probRow.Prob1WinsAway



                round_winningProb  = float(round_winningProb) +  getTotalProb_WinSeries ( p_H,p_A)


            AvgRound_winningProb =  float(round(round_winningProb / OpponentSeedsLen,5)) 

            if (r == 'R1'):
                totalWinProb = AvgRound_winningProb
            else:
                totalWinProb = totalWinProb * AvgRound_winningProb


            TeamWinningPercentagePerRound[thisSeedStr][r] = round(totalWinProb,5)
      
 

In [160]:
#Run this block to set up the dict TeamWinningPercentagePerRound for both East and West conference
TeamWinningPercentagePerRound = {}

getProbsToFinals('East')
getProbsToFinals('West')


In [161]:
TeamWinningPercentagePerRound

{'East1': {'R1': 0.99995, 'R2': 0.90132, 'R3': 0.76399},
 'East2': {'R1': 0.98736, 'R2': 0.81558, 'R3': 0.62451},
 'East3': {'R1': 0.90004, 'R2': 0.54929, 'R3': 0.3633},
 'East4': {'R1': 0.62984, 'R2': 0.34028, 'R3': 0.19261},
 'East5': {'R1': 0.37016, 'R2': 0.17397, 'R3': 0.07365},
 'East6': {'R1': 0.09996, 'R2': 0.03746, 'R3': 0.01267},
 'East7': {'R1': 0.01264, 'R2': 0.00239, 'R3': 0.00057},
 'East8': {'R1': 5e-05, 'R2': 0.0, 'R3': 0.0},
 'West1': {'R1': 0.99997, 'R2': 0.92447, 'R3': 0.80115},
 'West2': {'R1': 0.97958, 'R2': 0.80048, 'R3': 0.58202},
 'West3': {'R1': 0.86653, 'R2': 0.54715, 'R3': 0.3374},
 'West4': {'R1': 0.67494, 'R2': 0.36373, 'R3': 0.2157},
 'West5': {'R1': 0.32506, 'R2': 0.14336, 'R3': 0.06496},
 'West6': {'R1': 0.13347, 'R2': 0.04762, 'R3': 0.01559},
 'West7': {'R1': 0.02042, 'R2': 0.00397, 'R3': 0.00104},
 'West8': {'R1': 3e-05, 'R2': 0.0, 'R3': 0.0}}

In [232]:
## Function to get Revenue with given HomeTeam, AwayTeam, the round number and the number of games are playing 
def getRevenue(HomeTeam,AwayTeam, roundNumber, gameNumber):
    HomeTeamRev = Hypothetical_Playoff_Gate_Data_df.loc[(Hypothetical_Playoff_Gate_Data_df['HomeTeam'] == HomeTeam) ]
    AwayTeamRev = Hypothetical_Playoff_Gate_Data_df.loc[(Hypothetical_Playoff_Gate_Data_df['HomeTeam'] == AwayTeam) ]

    RoundX_GateStr = 'RoundX_Gate'

    revenue = 0
    if (gameNumber == 4):

        revenue =  float((HomeTeamRev[RoundX_GateStr.replace('X', str(roundNumber) )])*2) + float((AwayTeamRev[RoundX_GateStr.replace('X', str(roundNumber) )])*2)

    elif (gameNumber == 5):   

         revenue =  float((HomeTeamRev[RoundX_GateStr.replace('X', str(roundNumber) )])*3) + float((AwayTeamRev[RoundX_GateStr.replace('X', str(roundNumber) )])*2)

    elif (gameNumber == 6):   

         revenue =  float((HomeTeamRev[RoundX_GateStr.replace('X', str(roundNumber) )])*3) + float((AwayTeamRev[RoundX_GateStr.replace('X', str(roundNumber) )])*3)

    elif (gameNumber == 7):   

         revenue =  float((HomeTeamRev[RoundX_GateStr.replace('X', str(roundNumber) )])*4) + float((AwayTeamRev[RoundX_GateStr.replace('X', str(roundNumber) )])*3)

   
    #return '${:,.2f}'.format(revenue)
    return revenue

In [93]:
## Testing with 'West1' vs 'East1', playing 7 games, in round 4 (NBA Finals)
print (getRevenue('West1', 'East1', 4, 7))

$70,542,342.77


In [234]:
## Function to get Revenue for my team with given indication of my team is being a home team or away team, the round number and the number of games are playing 
def getMyTeamRevenue(MyTeam, homeOrAwayStr, roundNumber, gameNumber):
    
    
    MyTeam = Hypothetical_Playoff_Gate_Data_df.loc[(Hypothetical_Playoff_Gate_Data_df['HomeTeam'] == MyTeam) ]
  
    RoundX_GateStr = 'RoundX_Gate'

    revenue = 0
    if (gameNumber == 4):
        
        revenue =  float((MyTeam[RoundX_GateStr.replace('X', str(roundNumber) )])*2) 

    elif (gameNumber == 5):   
        
        if(homeOrAwayStr == 'home'):
                
            revenue =  float((MyTeam[RoundX_GateStr.replace('X', str(roundNumber) )])*3) 

        else:
            revenue =  float((MyTeam[RoundX_GateStr.replace('X', str(roundNumber) )])*2) 


    elif (gameNumber == 6):   

         revenue =  float((MyTeam[RoundX_GateStr.replace('X', str(roundNumber) )])*3) 

    elif (gameNumber == 7):   

        if(homeOrAwayStr == 'home'):
                
            revenue =  float((MyTeam[RoundX_GateStr.replace('X', str(roundNumber) )])*4) 

        else:
            revenue =  float((MyTeam[RoundX_GateStr.replace('X', str(roundNumber) )])*3) 


    #return '${:,.2f}'.format(revenue)
    return revenue

In [312]:
## Function to create Playoff Simulator dataframe with given team conference and seed. This function is leverging the outcomes from the previously created functions
def createPlayoffSimulatorDF(conference, seed):

    if(conference == 'East'):
        OppositConference = 'West'
    elif(conference == 'West'):
        OppositConference = 'East'

    df = pd.DataFrame()

    thisSeedStr = conference + str(seed)
    roundNumber = 1

    for r in Rounds:

            OpponentSeeds= roundMatchesDict[seed][r]
            OpponentSeedsLen = len(OpponentSeeds)

            HomeTeam = ''

            AwayTeam = ''
            
            myTeamStatus = ''

            round_winningProb = 0;

            for OpponentSeed in  OpponentSeeds:

                OpponentSeedStr = conference + str(OpponentSeed)

                p_H = 0
                p_A = 0

                if(seed > OpponentSeed):

                    HomeTeam = OpponentSeedStr
                    AwayTeam = thisSeedStr
                    myTeamStatus = 'Away'

                    probRow = win_probabilities_df.loc[(win_probabilities_df['Team1'] == OpponentSeedStr) & (win_probabilities_df['Team2'] == thisSeedStr)]

                    p_H = 1 - probRow.Prob1WinsHome
                    p_A = 1 - probRow.Prob1WinsAway


                else:

                    HomeTeam = thisSeedStr
                    AwayTeam = OpponentSeedStr
                    myTeamStatus = 'Home'

                    probRow = win_probabilities_df.loc[(win_probabilities_df['Team1'] == thisSeedStr) & (win_probabilities_df['Team2'] == OpponentSeedStr)]

                    p_H = probRow.Prob1WinsHome
                    p_A = probRow.Prob1WinsAway


                round_winningProb  =  getTotalProb_WinSeries (p_H,p_A)


                probOfEncounter = 0

                if(r == 'R1'):
                    probOfEncounter = 1
                else:
                    probOfEncounter = TeamWinningPercentagePerRound[OpponentSeedStr][r]

                probOfPlayingGame4 = getProb_In4( p_H,p_A,'win') + getProb_In4(p_H,p_A,'lose') 
                probOfPlayingGame5 = getProb_In5( p_H,p_A,'win') + getProb_In4(p_H,p_A,'lose')
                probOfPlayingGame6 = getProb_In6( p_H,p_A,'win') + getProb_In4(p_H,p_A,'lose')
                probOfPlayingGame7 = getProb_In7( p_H,p_A,'win') + getProb_In4(p_H,p_A,'lose')


                revOfPlayingGame4To7 = []

                TotalRevOfPlayingGame4To7 = []
                homeRevOfPlayingGame4To7 = []

                for i in range(4, 8):

                    TotalRevOfPlayingGame4To7.append(getRevenue(HomeTeam, AwayTeam, roundNumber, i)) 
                    homeRevOfPlayingGame4To7.append(getMyTeamRevenue(thisSeedStr, myTeamStatus, roundNumber, i)) 


    
                df = df.append({'Team': OpponentSeedStr, 
                                        'Round': r, 
                                        'ProbOfEncounter': float(round(probOfEncounter,5)),
                                        'ProbOfWin': float(round(round_winningProb,5)), 
                                        'ProbOfLose': float(round(1 - round_winningProb,5)), 
                                        'Avg_MyTeamRev': '${:,.2f}'.format(((homeRevOfPlayingGame4To7[0] + homeRevOfPlayingGame4To7[3]) / 2)),
                                        'Avg_TotalRev':  '${:,.2f}'.format(((TotalRevOfPlayingGame4To7[0] + TotalRevOfPlayingGame4To7[3]) / 2)), 
                                        'ProbOfPlaying_4G': float(round(probOfPlayingGame4,5)),
                                        'MyTeamRevOfPlaying_4G': '${:,.2f}'.format(homeRevOfPlayingGame4To7[0]),
                                        'TotalRevOfPlaying_4G': '${:,.2f}'.format(TotalRevOfPlayingGame4To7[0]),
                                        'ProbOfPlaying_5G': float(round(probOfPlayingGame5,5)),
                                        'MyTeamRevOfPlaying_5G': '${:,.2f}'.format(homeRevOfPlayingGame4To7[1]),
                                        'TotalRevOfPlaying_5G': '${:,.2f}'.format(TotalRevOfPlayingGame4To7[1]),
                                        'ProbOfPlaying_6G': float(round(probOfPlayingGame6,5)),
                                        'MyTeamRevOfPlaying_6G': '${:,.2f}'.format(homeRevOfPlayingGame4To7[2]),
                                        'TotalRevOfPlaying_6G': '${:,.2f}'.format(TotalRevOfPlayingGame4To7[2]),
                                        'ProbOfPlaying_7G': float(round(probOfPlayingGame7,5)),
                                        'MyTeamRevOfPlaying_7G': '${:,.2f}'.format(homeRevOfPlayingGame4To7[3]),
                                        'TotalRevOfPlaying_7G': '${:,.2f}'.format(TotalRevOfPlayingGame4To7[3])
                                        }, ignore_index=True)


            roundNumber = roundNumber + 1




    for OpponentSeed in Seeds:

        OpponentSeedStr =  OppositConference + str(OpponentSeed)

        HomeTeam = ''

        AwayTeam = ''
        
        myTeamStatus = ''

        round_winningProb = 0;

        p_H = 0
        p_A = 0

        if(seed > OpponentSeed or OppositConference == 'East'):

            
            HomeTeam = OpponentSeedStr
            AwayTeam = thisSeedStr
            
            if(conference != 'West'):
                myTeamStatus = 'Away'
                probRow = win_probabilities_df.loc[(win_probabilities_df['Team1'] == thisSeedStr) & (win_probabilities_df['Team2'] == OpponentSeedStr )]

                p_H = probRow.Prob1WinsHome
                p_A = probRow.Prob1WinsAway
                
            else:
                myTeamStatus = 'Home'
                probRow = win_probabilities_df.loc[(win_probabilities_df['Team1'] == OpponentSeedStr) & (win_probabilities_df['Team2'] == thisSeedStr)]

                p_H = 1 - probRow.Prob1WinsHome
                p_A = 1 - probRow.Prob1WinsAway
           


        elif(seed < OpponentSeed or seed == OpponentSeed ):
            

            HomeTeam = thisSeedStr
            AwayTeam = OpponentSeedStr
            
                     
            if(conference != 'West'):
                myTeamStatus = 'Away'
            else:
                myTeamStatus = 'Home'
       

            probRow = win_probabilities_df.loc[(win_probabilities_df['Team1'] == thisSeedStr) & (win_probabilities_df['Team2'] == OpponentSeedStr)]
            

            p_H = probRow.Prob1WinsHome
            p_A = probRow.Prob1WinsAway


        round_winningProb  =  getTotalProb_WinSeries (p_H,p_A)


        #probOfEncounter = The probability of Opponent Seed won all 3 rounds in their own conferences and got into the finals * The probability of my team won all 3 rounds in our own conferences and got into the finals 
        probOfEncounter = TeamWinningPercentagePerRound[thisSeedStr]['R3'] * TeamWinningPercentagePerRound[OpponentSeedStr]['R3']

        probOfPlayingGame4 = getProb_In4( p_H,p_A,'win') + getProb_In4(p_H,p_A,'lose') 
        probOfPlayingGame5 = getProb_In5( p_H,p_A,'win') + getProb_In4(p_H,p_A,'lose')
        probOfPlayingGame6 = getProb_In6( p_H,p_A,'win') + getProb_In4(p_H,p_A,'lose')
        probOfPlayingGame7 = getProb_In7( p_H,p_A,'win') + getProb_In4(p_H,p_A,'lose')


        TotalRevOfPlayingGame4To7 = []
        homeRevOfPlayingGame4To7 = []

        for i in range(4, 8):

            TotalRevOfPlayingGame4To7.append(getRevenue(HomeTeam, AwayTeam, roundNumber, i)) 
            homeRevOfPlayingGame4To7.append(getMyTeamRevenue(thisSeedStr, myTeamStatus, roundNumber, i)) 

  
            

        df = df.append({'Team': OpponentSeedStr, 
                                'Round': 'R4', 
                                'ProbOfEncounter': float(round(probOfEncounter,5)),
                                'ProbOfWin': float(round(round_winningProb,5)), 
                                'ProbOfLose': float(round(1 - round_winningProb,5)), 
                                'Avg_MyTeamRev': '${:,.2f}'.format(((homeRevOfPlayingGame4To7[0] + homeRevOfPlayingGame4To7[3]) / 2)),
                                'Avg_TotalRev':  '${:,.2f}'.format(((TotalRevOfPlayingGame4To7[0] + TotalRevOfPlayingGame4To7[3]) / 2)), 
                                'ProbOfPlaying_4G': float(round(probOfPlayingGame4,5)),
                                'MyTeamRevOfPlaying_4G': '${:,.2f}'.format(homeRevOfPlayingGame4To7[0]),
                                'TotalRevOfPlaying_4G': '${:,.2f}'.format(TotalRevOfPlayingGame4To7[0]),
                                'ProbOfPlaying_5G': float(round(probOfPlayingGame5,5)),
                                'MyTeamRevOfPlaying_5G': '${:,.2f}'.format(homeRevOfPlayingGame4To7[1]),
                                'TotalRevOfPlaying_5G': '${:,.2f}'.format(TotalRevOfPlayingGame4To7[1]),
                                'ProbOfPlaying_6G': float(round(probOfPlayingGame6,5)),
                                'MyTeamRevOfPlaying_6G': '${:,.2f}'.format(homeRevOfPlayingGame4To7[2]),
                                'TotalRevOfPlaying_6G': '${:,.2f}'.format(TotalRevOfPlayingGame4To7[2]),
                                'ProbOfPlaying_7G': float(round(probOfPlayingGame7,5)),
                                'MyTeamRevOfPlaying_7G': '${:,.2f}'.format(homeRevOfPlayingGame4To7[3]),
                                'TotalRevOfPlaying_7G': '${:,.2f}'.format(TotalRevOfPlayingGame4To7[3])
                                }, ignore_index=True)


        
      
     
    cols = ['Team' , 'Round','ProbOfEncounter', 'ProbOfWin', 'ProbOfLose', 'Avg_MyTeamRev', 'Avg_TotalRev', 'ProbOfPlaying_4G', 'MyTeamRevOfPlaying_4G', 'TotalRevOfPlaying_4G', 'ProbOfPlaying_5G','MyTeamRevOfPlaying_5G', 'TotalRevOfPlaying_5G', 'ProbOfPlaying_6G', 'MyTeamRevOfPlaying_6G', 'TotalRevOfPlaying_6G','ProbOfPlaying_7G', 'MyTeamRevOfPlaying_7G','TotalRevOfPlaying_7G']
    df = df[cols]
        

    return df

### Using the Playoff Simulator to create the dataframe

In [384]:
conference = 'West'
seed = 1

playoffSimulator_West_DF = createPlayoffSimulatorDF(conference, seed)
playoffSimulator_West_DF

Team Round  ProbOfEncounter  ProbOfWin  ProbOfLose   Avg_MyTeamRev  \
0   West8    R1          1.00000    0.99997     0.00003   $9,060,422.50   
1   West4    R2          0.36373    0.89096     0.10904  $12,322,174.60   
2   West5    R2          0.14336    0.95804     0.04196  $12,322,174.60   
3   West2    R3          0.58202    0.66957     0.33043  $21,440,583.80   
4   West3    R3          0.33740    0.80859     0.19141  $21,440,583.80   
5   West6    R3          0.01559    0.99019     0.00981  $21,440,583.80   
6   West7    R3          0.00104    0.99806     0.00194  $21,440,583.80   
7   East1    R4          0.61207    0.55332     0.44668  $28,305,774.95   
8   East2    R4          0.50033    0.55953     0.44047  $28,305,774.95   
9   East3    R4          0.29106    0.82661     0.17339  $28,305,774.95   
10  East4    R4          0.15431    0.84769     0.15231  $28,305,774.95   
11  East5    R4          0.05900    0.92764     0.07236  $28,305,774.95   
12  East6    R4          0.01015    0.98229     0.01771  $28,305,774.95   
13  East7    R4          0.00046    0.99737     0.00263  $28,305,774.95   
14  East8    R4          0.00000    1.00000     0.00000  $28,305,774.95   

      Avg_TotalRev  ProbOfPlaying_4G MyTeamRevOfPlaying_4G  \
0   $13,309,454.50           0.85753         $7,248,338.00   
1   $20,636,975.77           0.26136         $9,857,739.68   
2   $22,984,451.77           0.37540         $9,857,739.68   
3   $32,850,486.66           0.13863        $17,152,467.04   
4   $39,809,096.08           0.19296        $17,152,467.04   
5   $36,270,695.56           0.52281        $17,152,467.04   
6   $39,804,665.11           0.67186        $17,152,467.04   
7   $53,558,877.80           0.12690        $22,644,619.96   
8   $33,821,057.05           0.12738        $22,644,619.96   
9   $34,543,607.46           0.21293        $22,644,619.96   
10  $66,410,378.15           0.22696        $22,644,619.96   
11  $46,764,597.64           0.32181        $22,644,619.96   
12  $38,611,345.95           0.48654        $22,644,619.96   
13  $37,350,846.22           0.65965        $22,644,619.96   
14  $42,492,191.74           0.96047        $22,644,619.96   

   TotalRevOfPlaying_4G  ProbOfPlaying_5G MyTeamRevOfPlaying_5G  \
0         $9,197,896.00           0.13267         $7,248,338.00   
1        $14,538,032.68           0.31093         $9,857,739.68   
2        $16,416,013.48           0.33961         $9,857,739.68   
3        $22,849,895.92           0.23166        $17,152,467.04   
4        $28,416,783.46           0.27725        $17,152,467.04   
5        $25,586,063.04           0.33003        $17,152,467.04   
6        $28,413,238.68           0.25994        $17,152,467.04   
7        $39,480,021.86           0.19334        $22,644,619.96   
8        $26,321,474.69           0.19335        $22,644,619.96   
9        $26,803,174.96           0.26677        $22,644,619.96   
10       $48,047,688.76           0.28075        $22,644,619.96   
11       $34,950,501.76           0.30998        $22,644,619.96   
12       $29,515,000.62           0.31622        $22,644,619.96   
13       $28,674,667.48           0.25070        $22,644,619.96   
14       $32,102,231.15           0.03836        $22,644,619.96   

   TotalRevOfPlaying_5G  ProbOfPlaying_6G MyTeamRevOfPlaying_6G  \
0        $12,822,065.00           0.00898        $10,872,507.00   
1        $19,466,902.52           0.21228        $14,786,609.52   
2        $21,344,883.32           0.16983        $14,786,609.52   
3        $31,426,129.44           0.21474        $25,728,700.56   
4        $36,993,016.98           0.22218        $25,728,700.56   
5        $34,162,296.56           0.10312        $25,728,700.56   
6        $36,989,472.20           0.05509        $25,728,700.56   
7        $47,897,722.81           0.22444        $33,966,929.94   
8        $28,159,902.06           0.22625        $33,966,929.94   
9        $28,882,452.47           0.24330        $33,966,929.94   
1

In [357]:
conference = 'East'
seed = 1

playoffSimulator_East_DF = createPlayoffSimulatorDF(conference, seed)
playoffSimulator_East_DF

Team Round  ProbOfEncounter  ProbOfWin  ProbOfLose   Avg_MyTeamRev  \
0   East8    R1          1.00000    0.99995     0.00005   $6,891,115.00   
1   East4    R2          0.34028    0.86985     0.13015   $8,820,627.20   
2   East5    R2          0.17397    0.93290     0.06710   $8,820,627.20   
3   East2    R3          0.62451    0.61487     0.38513  $13,936,590.98   
4   East3    R3          0.36330    0.80694     0.19306  $13,936,590.98   
5   East6    R3          0.01267    0.96995     0.03005  $13,936,590.98   
6   East7    R3          0.00057    0.99876     0.00124  $13,936,590.98   
7   West1    R4          0.61207    0.44668     0.55332  $21,044,252.37   
8   West2    R4          0.44466    0.71315     0.28685  $21,044,252.37   
9   West3    R4          0.25777    0.78771     0.21229  $21,044,252.37   
10  West4    R4          0.16479    0.88931     0.11069  $21,044,252.37   
11  West5    R4          0.04963    0.93520     0.06480  $21,044,252.37   
12  West6    R4          0.01191    0.97743     0.02257  $21,044,252.37   
13  West7    R4          0.00079    0.99492     0.00508  $21,044,252.37   
14  West8    R4          0.00000    0.99968     0.00032  $21,044,252.37   

      Avg_TotalRev  ProbOfPlaying_4G MyTeamRevOfPlaying_4G  \
0   $11,862,533.00           0.84875         $5,512,892.00   
1   $21,980,760.39           0.23880         $7,056,501.76   
2   $16,951,620.29           0.31453         $7,056,501.76   
3   $19,632,813.23           0.13107        $11,149,272.78   
4   $19,799,763.46           0.19396        $11,149,272.78   
5   $22,264,538.74           0.41798        $11,149,272.78   
6   $21,319,981.97           0.69869        $11,149,272.78   
7   $53,558,877.80           0.12690        $16,835,401.90   
8   $37,288,921.36           0.15195        $16,835,401.90   
9   $46,373,696.13           0.18369        $16,835,401.90   
10  $37,932,016.59           0.25595        $16,835,401.90   
11  $46,649,471.12           0.31746        $16,835,401.90   
12  $41,066,095.35           0.44703        $16,835,401.90   
13  $49,886,040.80           0.58313        $16,835,401.90   
14  $33,022,130.47           0.77329        $16,835,401.90   

   TotalRevOfPlaying_4G  ProbOfPlaying_5G MyTeamRevOfPlaying_5G  \
0         $8,387,448.00           0.13956         $5,512,892.00   
1        $16,173,307.96           0.30269         $7,056,501.76   
2        $12,149,995.88           0.33667         $7,056,501.76   
3        $13,476,396.03           0.21412        $11,149,272.78   
4        $13,609,956.21           0.27378        $11,149,272.78   
5        $15,581,776.43           0.33579        $11,149,272.78   
6        $14,826,131.02           0.24618        $11,149,272.78   
7        $39,480,021.86           0.18447        $16,835,401.90   
8        $26,464,056.71           0.24228        $16,835,401.90   
9        $33,731,876.52           0.26588        $16,835,401.90   
10       $26,978,532.89           0.31462        $16,835,401.90   
11       $33,952,496.52           0.33894        $16,835,401.90   
12       $29,485,795.90           0.33697        $16,835,401.90   
13       $36,541,752.26           0.30872        $16,835,401.90   
14       $23,050,624.00           0.19802        $16,835,401.90   

   TotalRevOfPlaying_5G  ProbOfPlaying_6G MyTeamRevOfPlaying_6G  \
0        $11,143,894.00           0.01065         $8,269,338.00   
1        $19,701,558.84           0.21630        $10,584,752.64   
2        $15,678,246.76           0.18794        $10,584,752.64   
3        $19,051,032.42           0.21654        $16,723,909.17   
4        $19,184,592.60           0.22564        $16,723,909.17   
5        $21,156,412.82           0.15621        $16,723,909.17   
6        $20,400,767.41           0.04533        $16,723,909.17   
7        $47,897,722.81           0.21453        $25,253,102.85   
8        $34,881,757.66           0.22125        $25,253,102.85   
9        $42,149,577.47           0.22663        $25,253,102.85   
1

In [296]:
#function to get expected revenues from own conference
def getConferenceCumulativeRev(conference,seed,playoffSimulatorDF):

    thisSeedStr = conference + str(seed)
    
    df = pd.DataFrame()
    
    
    cumulativeMyTeamRev = 0
    cumulativeTotalRev = 0
    
    for i in range(0,7):

        cumulativeMyTeamRev = float(cumulativeMyTeamRev) + playoffSimulatorDF.loc[[i]]['ProbOfEncounter']  * playoffSimulatorDF.loc[[i]]['Avg_MyTeamRev'].replace( '[\$,)]','', regex=True).astype(float)
        cumulativeTotalRev = float(cumulativeTotalRev) + playoffSimulatorDF.loc[[i]]['ProbOfEncounter']  * playoffSimulatorDF.loc[[i]]['Avg_TotalRev'].replace( '[\$,)]','', regex=True).astype(float)
    
    
    
    df = df.append({'Conference': conference, 
                            'Seed': seed, 
                            'ProbOfEarningThisRevenue': TeamWinningPercentagePerRound[thisSeedStr]['R3'],
                            'Cumulative_MyTeamRev': '${:,.2f}'.format(float(cumulativeMyTeamRev)), 
                            'Cumulative_TotalRev': '${:,.2f}'.format(float(cumulativeTotalRev))    
                            }, ignore_index=True)

     
    cols = ['Conference' , 'Seed','ProbOfEarningThisRevenue', 'Cumulative_MyTeamRev', 'Cumulative_TotalRev']
    df = df[cols]
        

    return df

In [297]:
getConferenceCumulativeRev('West',1,playoffSimulator_West_DF)

Conference  Seed  ProbOfEarningThisRevenue Cumulative_MyTeamRev  \
0       West   1.0                   0.80115       $35,378,332.48   

  Cumulative_TotalRev  
0      $57,268,878.96

In [293]:
getConferenceCumulativeRev('East',1,playoffSimulator_East_DF)

Conference  Seed  ProbOfEarningThisRevenue Cumulative_MyTeamRev  \
0       East   1.0                   0.76399       $25,378,346.94   

  Cumulative_TotalRev  
0      $42,039,605.88

In [367]:
## function to get cumuated revenue after finals with given my team conference, my team seed, opponent team conference and opponent team seed
def getCumulatedRevenueAfterFinals(myTeamConference,myTeamSeed,OpponentTeamConference,OpponentTeamSeed):
    df = pd.DataFrame()

    thisSeedStr = myTeamConference + str(myTeamSeed)
    OpponentSeedStr = OpponentTeamConference + str(OpponentTeamSeed)



    myTeamPlayoffSimulator_DF = createPlayoffSimulatorDF(myTeamConference, myTeamSeed)

    opponentTeamPlayoffSimulator_DF = createPlayoffSimulatorDF(OpponentTeamConference, OpponentTeamSeed)


    myTeamConference_DF = getConferenceCumulativeRev(myTeamConference,myTeamSeed,myTeamPlayoffSimulator_DF)

    opponentTeamConference_DF = getConferenceCumulativeRev(OpponentTeamConference,OpponentTeamSeed,opponentTeamPlayoffSimulator_DF)


    cumulative_MyTeamRev = float(myTeamConference_DF.Cumulative_MyTeamRev.replace( '[\$,)]','', regex=True).astype(float)) + float(myTeamPlayoffSimulator_DF.loc[(myTeamPlayoffSimulator_DF['Team'] == OpponentSeedStr)].Avg_MyTeamRev.replace( '[\$,)]','', regex=True).astype(float))

    cumulative_TotalRev =  float(opponentTeamConference_DF.Cumulative_TotalRev.replace( '[\$,)]','', regex=True).astype(float))+ float(myTeamConference_DF.Cumulative_TotalRev.replace( '[\$,)]','', regex=True).astype(float)) + float(myTeamPlayoffSimulator_DF.loc[(myTeamPlayoffSimulator_DF['Team'] == OpponentSeedStr)].Avg_TotalRev.replace( '[\$,)]','', regex=True).astype(float))


    df = df.append({'MyTeam': thisSeedStr, 
                        'OpponentTeam': OpponentSeedStr, 
                        'Cumulative_MyTeamRev': '${:,.2f}'.format(float(cumulative_MyTeamRev)), 
                        'Cumulative_BothTeamsTotalRev': '${:,.2f}'.format(float(cumulative_TotalRev))
                        }, ignore_index=True)


    cols = ['MyTeam' , 'OpponentTeam', 'Cumulative_MyTeamRev', 'Cumulative_BothTeamsTotalRev']

    df = df[cols]


    return df


In [382]:
myTeamConference = 'East'
myTeamSeed = 1
OpponentTeamConference = 'West'
OpponentTeamSeed = 1

getCumulatedRevenueAfterFinals(myTeamConference,myTeamSeed,OpponentTeamConference,OpponentTeamSeed)

MyTeam OpponentTeam Cumulative_MyTeamRev Cumulative_BothTeamsTotalRev
0  East1        West1       $46,422,599.31              $152,867,362.64

In [383]:
myTeamConference = 'West'
myTeamSeed = 1
OpponentTeamConference = 'East'
OpponentTeamSeed = 1

getCumulatedRevenueAfterFinals(myTeamConference,myTeamSeed,OpponentTeamConference,OpponentTeamSeed)

MyTeam OpponentTeam Cumulative_MyTeamRev Cumulative_BothTeamsTotalRev
0  West1        East1       $63,684,107.43              $152,867,362.64

## c) How do you propose to present a solution to the team? You want to ensure the team understands that there is a relatively wide distribution of possible outcomes. What is the solution?

I propose to present the solution by given the data for the wide distribution of possible outcomes so the team can have the compresensive overview of what to expect from the forcasting outcomes. 

I create the Playoff Simulator as below to let the team to input our team's conference and seed number, then the Playoff Simulator will layout the teams that our team may face in each round with the probability of facing, wining , losing, playing 4 ,5 ,6 or 7 games. Along with the revenues that will be generate for each possible combinations for both our team and our team and opponent team combineded. 

This can allow the team to have the sense of at worst or at the most, how much revenue they can earn through out the different commination of games along with the probability of earning those revenues.

In [390]:
conference = 'West'
seed = 1

playoffSimulator_West_DF = createPlayoffSimulatorDF(conference, seed)
playoffSimulator_West_DF

Team Round  ProbOfEncounter  ProbOfWin  ProbOfLose   Avg_MyTeamRev  \
0   West8    R1          1.00000    0.99997     0.00003   $9,060,422.50   
1   West4    R2          0.36373    0.89096     0.10904  $12,322,174.60   
2   West5    R2          0.14336    0.95804     0.04196  $12,322,174.60   
3   West2    R3          0.58202    0.66957     0.33043  $21,440,583.80   
4   West3    R3          0.33740    0.80859     0.19141  $21,440,583.80   
5   West6    R3          0.01559    0.99019     0.00981  $21,440,583.80   
6   West7    R3          0.00104    0.99806     0.00194  $21,440,583.80   
7   East1    R4          0.61207    0.55332     0.44668  $28,305,774.95   
8   East2    R4          0.50033    0.55953     0.44047  $28,305,774.95   
9   East3    R4          0.29106    0.82661     0.17339  $28,305,774.95   
10  East4    R4          0.15431    0.84769     0.15231  $28,305,774.95   
11  East5    R4          0.05900    0.92764     0.07236  $28,305,774.95   
12  East6    R4          0.01015    0.98229     0.01771  $28,305,774.95   
13  East7    R4          0.00046    0.99737     0.00263  $28,305,774.95   
14  East8    R4          0.00000    1.00000     0.00000  $28,305,774.95   

      Avg_TotalRev  ProbOfPlaying_4G MyTeamRevOfPlaying_4G  \
0   $13,309,454.50           0.85753         $7,248,338.00   
1   $20,636,975.77           0.26136         $9,857,739.68   
2   $22,984,451.77           0.37540         $9,857,739.68   
3   $32,850,486.66           0.13863        $17,152,467.04   
4   $39,809,096.08           0.19296        $17,152,467.04   
5   $36,270,695.56           0.52281        $17,152,467.04   
6   $39,804,665.11           0.67186        $17,152,467.04   
7   $53,558,877.80           0.12690        $22,644,619.96   
8   $33,821,057.05           0.12738        $22,644,619.96   
9   $34,543,607.46           0.21293        $22,644,619.96   
10  $66,410,378.15           0.22696        $22,644,619.96   
11  $46,764,597.64           0.32181        $22,644,619.96   
12  $38,611,345.95           0.48654        $22,644,619.96   
13  $37,350,846.22           0.65965        $22,644,619.96   
14  $42,492,191.74           0.96047        $22,644,619.96   

   TotalRevOfPlaying_4G  ProbOfPlaying_5G MyTeamRevOfPlaying_5G  \
0         $9,197,896.00           0.13267         $7,248,338.00   
1        $14,538,032.68           0.31093         $9,857,739.68   
2        $16,416,013.48           0.33961         $9,857,739.68   
3        $22,849,895.92           0.23166        $17,152,467.04   
4        $28,416,783.46           0.27725        $17,152,467.04   
5        $25,586,063.04           0.33003        $17,152,467.04   
6        $28,413,238.68           0.25994        $17,152,467.04   
7        $39,480,021.86           0.19334        $22,644,619.96   
8        $26,321,474.69           0.19335        $22,644,619.96   
9        $26,803,174.96           0.26677        $22,644,619.96   
10       $48,047,688.76           0.28075        $22,644,619.96   
11       $34,950,501.76           0.30998        $22,644,619.96   
12       $29,515,000.62           0.31622        $22,644,619.96   
13       $28,674,667.48           0.25070        $22,644,619.96   
14       $32,102,231.15           0.03836        $22,644,619.96   

   TotalRevOfPlaying_5G  ProbOfPlaying_6G MyTeamRevOfPlaying_6G  \
0        $12,822,065.00           0.00898        $10,872,507.00   
1        $19,466,902.52           0.21228        $14,786,609.52   
2        $21,344,883.32           0.16983        $14,786,609.52   
3        $31,426,129.44           0.21474        $25,728,700.56   
4        $36,993,016.98           0.22218        $25,728,700.56   
5        $34,162,296.56           0.10312        $25,728,700.56   
6        $36,989,472.20           0.05509        $25,728,700.56   
7        $47,897,722.81           0.22444        $33,966,929.94   
8        $28,159,902.06           0.22625        $33,966,929.94   
9        $28,882,452.47           0.24330        $33,966,929.94   
1

## d) What is the probability that West1 meets East1 in the Finals? What is the expected total Playoffs gate revenue if West1 meets East1 in the Finals? What if West1 does not meet East1 in the Finals? How does your solution change if – independent of all other series – East5 defeats East4 in the First Round? You don’t know how many games the series went; you just know the outcome.

#### What is the probability that West1 meets East1 in the Finals? 

In [385]:
conference = 'West'
seed = 1

playoffSimulator_West_DF = createPlayoffSimulatorDF(conference, seed)
playoffSimulator_West_DF

Team Round  ProbOfEncounter  ProbOfWin  ProbOfLose   Avg_MyTeamRev  \
0   West8    R1          1.00000    0.99997     0.00003   $9,060,422.50   
1   West4    R2          0.36373    0.89096     0.10904  $12,322,174.60   
2   West5    R2          0.14336    0.95804     0.04196  $12,322,174.60   
3   West2    R3          0.58202    0.66957     0.33043  $21,440,583.80   
4   West3    R3          0.33740    0.80859     0.19141  $21,440,583.80   
5   West6    R3          0.01559    0.99019     0.00981  $21,440,583.80   
6   West7    R3          0.00104    0.99806     0.00194  $21,440,583.80   
7   East1    R4          0.61207    0.55332     0.44668  $28,305,774.95   
8   East2    R4          0.50033    0.55953     0.44047  $28,305,774.95   
9   East3    R4          0.29106    0.82661     0.17339  $28,305,774.95   
10  East4    R4          0.15431    0.84769     0.15231  $28,305,774.95   
11  East5    R4          0.05900    0.92764     0.07236  $28,305,774.95   
12  East6    R4          0.01015    0.98229     0.01771  $28,305,774.95   
13  East7    R4          0.00046    0.99737     0.00263  $28,305,774.95   
14  East8    R4          0.00000    1.00000     0.00000  $28,305,774.95   

      Avg_TotalRev  ProbOfPlaying_4G MyTeamRevOfPlaying_4G  \
0   $13,309,454.50           0.85753         $7,248,338.00   
1   $20,636,975.77           0.26136         $9,857,739.68   
2   $22,984,451.77           0.37540         $9,857,739.68   
3   $32,850,486.66           0.13863        $17,152,467.04   
4   $39,809,096.08           0.19296        $17,152,467.04   
5   $36,270,695.56           0.52281        $17,152,467.04   
6   $39,804,665.11           0.67186        $17,152,467.04   
7   $53,558,877.80           0.12690        $22,644,619.96   
8   $33,821,057.05           0.12738        $22,644,619.96   
9   $34,543,607.46           0.21293        $22,644,619.96   
10  $66,410,378.15           0.22696        $22,644,619.96   
11  $46,764,597.64           0.32181        $22,644,619.96   
12  $38,611,345.95           0.48654        $22,644,619.96   
13  $37,350,846.22           0.65965        $22,644,619.96   
14  $42,492,191.74           0.96047        $22,644,619.96   

   TotalRevOfPlaying_4G  ProbOfPlaying_5G MyTeamRevOfPlaying_5G  \
0         $9,197,896.00           0.13267         $7,248,338.00   
1        $14,538,032.68           0.31093         $9,857,739.68   
2        $16,416,013.48           0.33961         $9,857,739.68   
3        $22,849,895.92           0.23166        $17,152,467.04   
4        $28,416,783.46           0.27725        $17,152,467.04   
5        $25,586,063.04           0.33003        $17,152,467.04   
6        $28,413,238.68           0.25994        $17,152,467.04   
7        $39,480,021.86           0.19334        $22,644,619.96   
8        $26,321,474.69           0.19335        $22,644,619.96   
9        $26,803,174.96           0.26677        $22,644,619.96   
10       $48,047,688.76           0.28075        $22,644,619.96   
11       $34,950,501.76           0.30998        $22,644,619.96   
12       $29,515,000.62           0.31622        $22,644,619.96   
13       $28,674,667.48           0.25070        $22,644,619.96   
14       $32,102,231.15           0.03836        $22,644,619.96   

   TotalRevOfPlaying_5G  ProbOfPlaying_6G MyTeamRevOfPlaying_6G  \
0        $12,822,065.00           0.00898        $10,872,507.00   
1        $19,466,902.52           0.21228        $14,786,609.52   
2        $21,344,883.32           0.16983        $14,786,609.52   
3        $31,426,129.44           0.21474        $25,728,700.56   
4        $36,993,016.98           0.22218        $25,728,700.56   
5        $34,162,296.56           0.10312        $25,728,700.56   
6        $36,989,472.20           0.05509        $25,728,700.56   
7        $47,897,722.81           0.22444        $33,966,929.94   
8        $28,159,902.06           0.22625        $33,966,929.94   
9        $28,882,452.47           0.24330        $33,966,929.94   
1

By running our playoffSimulator by setting conference to 'West' and seed to 1, we see that for West1, the probability to face East1 in the NBA finals is 0.61207


#### What is the expected total Playoffs gate revenue if West1 meets East1 in the Finals?

In [386]:
myTeamConference = 'West'
myTeamSeed = 1
OpponentTeamConference = 'East'
OpponentTeamSeed = 1

getCumulatedRevenueAfterFinals(myTeamConference,myTeamSeed,OpponentTeamConference,OpponentTeamSeed)

MyTeam OpponentTeam Cumulative_MyTeamRev Cumulative_BothTeamsTotalRev
0  West1        East1       $63,684,107.43              $152,867,362.64

By running our function to get the cumulated revenue after Finals (Round 4), we see that West1 can earn \$ 63,684,107.43  alone. And combine with the opponent team East 1, their cumulated revenue is \$152,867,362.64

This is partially cummulated with the outcome of the following function to see how much money West1 can earn if it won all 3 rounds in the Western Conference.

In [388]:
getConferenceCumulativeRev(conference,seed,playoffSimulator_West_DF)

Conference  Seed  ProbOfEarningThisRevenue Cumulative_MyTeamRev  \
0       West   1.0                   0.80115       $35,378,332.48   

  Cumulative_TotalRev  
0      $57,268,878.96

#### What if West1 does not meet East1 in the Finals? How does your solution change if – independent of all other series – East5 defeats East4 in the First Round? You don’t know how many games the series went; you just know the outcome.¶

As the PlayoffSimulator will give you all the probabilities and outcomes of all the possible facing opponents based on their own probabilities of making to the specifit round with the expected probabilities and outcomes that calculated based on their own combinations of mataches with their own opponents.

By looking at the listed expected values, the team can immediately see the possible revenues and the probabilities of earning for facing either East 1 or East 8 in the NBA Finals. 

We may not know how many games the series went, but by listing all the possible outcomes and probabilistic thinking, we can immediate get the expected revenues and the likelihood of getting the expected revenues.  


In [392]:
conference = 'West'
seed = 1

playoffSimulator_West_DF = createPlayoffSimulatorDF(conference, seed)
playoffSimulator_West_DF

Team Round  ProbOfEncounter  ProbOfWin  ProbOfLose   Avg_MyTeamRev  \
0   West8    R1          1.00000    0.99997     0.00003   $9,060,422.50   
1   West4    R2          0.36373    0.89096     0.10904  $12,322,174.60   
2   West5    R2          0.14336    0.95804     0.04196  $12,322,174.60   
3   West2    R3          0.58202    0.66957     0.33043  $21,440,583.80   
4   West3    R3          0.33740    0.80859     0.19141  $21,440,583.80   
5   West6    R3          0.01559    0.99019     0.00981  $21,440,583.80   
6   West7    R3          0.00104    0.99806     0.00194  $21,440,583.80   
7   East1    R4          0.61207    0.55332     0.44668  $28,305,774.95   
8   East2    R4          0.50033    0.55953     0.44047  $28,305,774.95   
9   East3    R4          0.29106    0.82661     0.17339  $28,305,774.95   
10  East4    R4          0.15431    0.84769     0.15231  $28,305,774.95   
11  East5    R4          0.05900    0.92764     0.07236  $28,305,774.95   
12  East6    R4          0.01015    0.98229     0.01771  $28,305,774.95   
13  East7    R4          0.00046    0.99737     0.00263  $28,305,774.95   
14  East8    R4          0.00000    1.00000     0.00000  $28,305,774.95   

      Avg_TotalRev  ProbOfPlaying_4G MyTeamRevOfPlaying_4G  \
0   $13,309,454.50           0.85753         $7,248,338.00   
1   $20,636,975.77           0.26136         $9,857,739.68   
2   $22,984,451.77           0.37540         $9,857,739.68   
3   $32,850,486.66           0.13863        $17,152,467.04   
4   $39,809,096.08           0.19296        $17,152,467.04   
5   $36,270,695.56           0.52281        $17,152,467.04   
6   $39,804,665.11           0.67186        $17,152,467.04   
7   $53,558,877.80           0.12690        $22,644,619.96   
8   $33,821,057.05           0.12738        $22,644,619.96   
9   $34,543,607.46           0.21293        $22,644,619.96   
10  $66,410,378.15           0.22696        $22,644,619.96   
11  $46,764,597.64           0.32181        $22,644,619.96   
12  $38,611,345.95           0.48654        $22,644,619.96   
13  $37,350,846.22           0.65965        $22,644,619.96   
14  $42,492,191.74           0.96047        $22,644,619.96   

   TotalRevOfPlaying_4G  ProbOfPlaying_5G MyTeamRevOfPlaying_5G  \
0         $9,197,896.00           0.13267         $7,248,338.00   
1        $14,538,032.68           0.31093         $9,857,739.68   
2        $16,416,013.48           0.33961         $9,857,739.68   
3        $22,849,895.92           0.23166        $17,152,467.04   
4        $28,416,783.46           0.27725        $17,152,467.04   
5        $25,586,063.04           0.33003        $17,152,467.04   
6        $28,413,238.68           0.25994        $17,152,467.04   
7        $39,480,021.86           0.19334        $22,644,619.96   
8        $26,321,474.69           0.19335        $22,644,619.96   
9        $26,803,174.96           0.26677        $22,644,619.96   
10       $48,047,688.76           0.28075        $22,644,619.96   
11       $34,950,501.76           0.30998        $22,644,619.96   
12       $29,515,000.62           0.31622        $22,644,619.96   
13       $28,674,667.48           0.25070        $22,644,619.96   
14       $32,102,231.15           0.03836        $22,644,619.96   

   TotalRevOfPlaying_5G  ProbOfPlaying_6G MyTeamRevOfPlaying_6G  \
0        $12,822,065.00           0.00898        $10,872,507.00   
1        $19,466,902.52           0.21228        $14,786,609.52   
2        $21,344,883.32           0.16983        $14,786,609.52   
3        $31,426,129.44           0.21474        $25,728,700.56   
4        $36,993,016.98           0.22218        $25,728,700.56   
5        $34,162,296.56           0.10312        $25,728,700.56   
6        $36,989,472.20           0.05509        $25,728,700.56   
7        $47,897,722.81           0.22444        $33,966,929.94   
8        $28,159,902.06           0.22625        $33,966,929.94   
9        $28,882,452.47           0.24330        $33,966,929.94   
1